enginetest/evaluation.go

```diff
@@ -666,13 +666,23 @@ func checkResults(
 			for i, val := range widenedRow {
 				if d, ok := val.(decimal.Decimal); ok {
 					widenedRow[i] = d.StringFixed(d.Exponent() * -1)
+				} else if idxVal, ok := val.(uint64); ok && !isServerEngine && sch != nil && len(sch) == len(widenedRow) {
```

Comment:
Test results look great! I hard a hard time reading this block of code though. The If/For/For/If/etc nesting is deep and kind of a mental workout to digest all the cases. Would be helpful to tidy up this to make it a little easier to read, but otherwise this all look like a great change! 🙌 

Will this require a coordinated change with Dolt test cases that use sets/enums, too?

Reply:
Ah yes, will check on Dolt side 🙂